pkg/core/native/designate.go

```diff
@@ -120,14 +121,37 @@ func newDesignate(p2pSigExtensionsEnabled bool) *Designate {
 	return s
 }
 
-// Initialize initializes Oracle contract.
+// Initialize initializes Designation contract. It is called once at native Management's OnPersist
+// at the genesis block, and we can't properly fill the cache at this point, as there are no roles
+// data in the storage.
 func (s *Designate) Initialize(ic *interop.Context) error {
-	cache := &DesignationCache{}
-	cache.rolesChangedFlag = true
+	cache := &DesignationCache{
+		rolesChangedFlag: false,
```

Comment:
I did this for the rest of contracts (NEO is still in progress). But we need this particular flag for Designation and it can't be dropped here, although we can rename it. The reason is that Designation has to notify the services about new node roles, and we can't perform notifications in-place, because transaction that changes node roles may fail. Thus, we should notify services in PostPersist where the final roles changes are present. Flag `rolesChangedFlag` now has another meaning and is responsible for notifications. The rest of Designation cache acts like the other updated caches and is always valid and up-to-date.